pulp_2to3_migration/tests/functional/test_migration_plan.py

```diff
@@ -0,0 +1,150 @@
+
+from pulp_smash import cli
+from pulp_smash import config as smash_config
+from pulp_smash.pulp3.bindings import monitor_task, monitor_task_group, PulpTaskError
+
+from .constants import TRUNCATE_TABLES_QUERY_BASH
+
+
+class TestMigrationPlan(unittest.TestCase):
+    """Test the Migration Plan creation and validation"""
+
+    smash_cfg = smash_config.get_config()
+    smash_cli_client = cli.Client(smash_cfg)
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Create all the client instances needed to communicate with Pulp.
+        """
+        configuration = Configuration()
+        configuration.username = 'admin'
+        configuration.password = 'password'
+        configuration.host = 'http://pulp'
+        configuration.safe_chars_for_path_param = '/'
+
+        core_client = CoreApiClient(configuration)
+        file_client = FileApiClient(configuration)
+        migration_client = MigrationApiClient(configuration)
+
+        # Create api clients for all resource types
+        cls.file_repo_api = RepositoriesFileApi(file_client)
+        cls.file_repo_versions_api = RepositoriesFileVersionsApi(file_client)
+        cls.file_content_api = ContentFilesApi(file_client)
+        cls.tasks_api = TasksApi(core_client)
+        cls.migration_plans_api = MigrationPlansApi(migration_client)
+
+        set_pulp2_snapshot(name='file_base_4repos')
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Clean up the database after the set of tests is run.
+        """
+        cmd = get_psql_smash_cmd(TRUNCATE_TABLES_QUERY_BASH)
+        cls.smash_cli_client.run(cmd, sudo=True)
+
+    def _load_and_run(self, plan, run_params={}):
+        """Load and run a migration plan."""
+        mp = self.migration_plans_api.create({'plan': plan})
+        mp_run_response = self.migration_plans_api.run(mp.pulp_href, run_params)
+        task = monitor_task(mp_run_response.task)
+        # to ensure that migration fully finished and further tests won't collide with it
+        monitor_task_group(task.task_group)
+        return task
+
+    def _do_test_parallel(self, plan, outcome):
+        """Test that there were multiple tasks running in parallel as a part of a task group."""
+        mp = self.migration_plans_api.create({'plan': plan})
+        mp_run_response = self.migration_plans_api.run(mp.pulp_href, {})
+        task = monitor_task(mp_run_response.task)
+        group = monitor_task_group(task.task_group)
+        self.assertEqual(group.completed, outcome['task_count'])
+
+    def test_load_simple_plan(self):
+        """Test that a simple Migration Plan can be loaded and run."""
+        task = self._load_and_run(plans.SIMPLE_PLAN_TEST.plan)
+        self.assertEqual(task.state, "completed")
+
+    def test_load_complex_plan(self):
+        """Test that a complex Migration Plan can be loaded and run."""
+        task = self._load_and_run(plans.COMPLEX_PLAN_TEST.plan)
+        self.assertEqual(task.state, "completed")
```

Comment:
This check isn't necessary, ```monitor_task()``` checks the state and raises an exception if it didn't end up as completed.